arcgis-ios-sdk-samples/Edit data/Edit features with feature-linked annotation/EditFeaturesWithFeatureLinkedAnnotationViewController.swift

```diff
@@ -0,0 +1,228 @@
+            }
+        }
+    }
+    
+    // MARK: - AGSGeoViewTouchDelegate
+    // Select the nearest feature or move the point or polyline vertex to the given screen point.
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // If a feature hasn't been selected.
+        if selectedFeature == nil {
+            selectFeature(screenPoint: screenPoint)
+        } else {
+            // Move the feature.
+            if selectedFeatureIsPolyline {
+                movePolylineVertex(mapPoint: mapPoint)
+            } else {
+                movePoint(mapPoint: mapPoint)
+            }
+        }
+    }
+    
+    func selectFeature(screenPoint: CGPoint) {
+        // Clear any previously selected features.
+        clearSelection()
+        selectedFeature = nil
+        
+        // Identify across all layers.
+        mapView.identifyLayers(atScreenPoint: screenPoint, tolerance: 10.0, returnPopupsOnly: false) { (results: [AGSIdentifyLayerResult]?, error: Error?) in
+            if let error = error {
+                self.presentAlert(error: error)
+                return
+            }
+            results?.forEach { (result) in
+                //Get a reference to the identified feature layer and feature.
+                guard let featureLayer = result.layerContent as? AGSFeatureLayer, let selectedFeature = result.geoElements[0] as? AGSFeature else { return }
+                switch selectedFeature.geometry?.geometryType {
+                case .point:
+                    // If the selected feature is a point, prompt the edit attributes alert.
+                    self.showEditableAttributes(selectedFeature: selectedFeature)
+                case .polyline:
+                    let polyline = selectedFeature.geometry as! AGSPolyline
+                    let polylineArray = polyline.parts.array()
+                    // If the selected feature is a polyline with any part containing more than one segment (i.e. a curve), present an alert.
+                    for part in polylineArray {
+                        if part.pointCount > 2 {
+                            // Set the value to nil.
+                            self.selectedFeature = nil
+                            // Present the alert.
+                            self.presentAlert(title: "Make a different selection", message: "Select straight (single segment) polylines only.")
+                            return
+                        }
+                    }
+                    // If the selected feature is a valid polyline, set the value to true.
+                    self.selectedFeatureIsPolyline = true
+                default:
+                    return
+                }
+                // Select the feature.
+                featureLayer.select(selectedFeature)
+                // Set the feature globally.
+                self.selectedFeature = selectedFeature
```

Comment:
The results array is being iterated, so it seems like there's the potential for multiple features to be selected. Should `selectedFeature` actually be a set?

Reply:
Only one feature is to be selected but the results are narrowed down on line 73. Is there another method to select the desired feature that avoids iterating through the array? I've tried a few methods of getting the first result but they didn't work out.

Reply:
The thing with `forEach` is that there's no way to stop iterating. The provided closure will be called for each element in the sequence. I haven't tested it, but it seems like this would work:

```swift
if let error = error {
    ...
} else if let results = results,
          let firstResult = results.first(where: { $0.layerContent is AGSFeature && !$0.geoElements.isEmpty }) {
    let featureLayer = firstResult.layerContent as! AGSFeatureLayer
    let selectedFeature = firstResult.geoElements.first as! AGSFeature
    ...
}
```

Reply:
Nice catch! I noticed this for another forEach earlier but missed this one...

Reply:
@philium I tested the closure you provided but it didn't work. I found that the desired feature isn't always the first result. So far, I've seen that it could also be the second result but I can't confirm that the desired features are exclusive to only the first and second results.

Reply:
Maybe bad but simple suggestion: revert forEach back to for ... in, to allow return statement cut off the loop.